usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -577,3 +585,93 @@ def fetch_naics_hierarchy(naics: str) -> dict:
+            select fyp from closed_periods
+        )
+    ),
+    eligible_file_c_records as (
+        select
+            faba.award_id,
+            faba.gross_outlay_amount_by_award_cpe,
+            faba.transaction_obligated_amount,
+            faba.disaster_emergency_fund_code,
+            s.reporting_fiscal_year,
+            s.reporting_fiscal_period
+        from financial_accounts_by_awards faba
+        inner join eligible_submissions s on s.submission_id = faba.submission_id
+    ),
+    fy_final_balances as (
+        -- Rule: If a balance is not zero at the end of the year, it must be reported in the
+        -- final period's submission (month or quarter), otherwise assume it to be zero
+        select faba.award_id, sum(faba.gross_outlay_amount_by_award_cpe) as prior_fys_outlay,
+            faba.disaster_emergency_fund_code
+        from eligible_file_c_records faba
+        group by
+            faba.award_id,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code
+        having faba.reporting_fiscal_period = 12
+        and sum(faba.gross_outlay_amount_by_award_cpe) > 0
+    ),
+     current_fy_balance as (
+        select
+            faba.award_id,
+            faba.reporting_fiscal_year,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code,
+            sum(faba.gross_outlay_amount_by_award_cpe) as current_fy_outlay
+        from eligible_file_c_records faba
+        group by
+            faba.award_id,
+            faba.reporting_fiscal_year,
+            faba.reporting_fiscal_period,
+            faba.disaster_emergency_fund_code
+        having concat(faba.reporting_fiscal_year::text, lpad(faba.reporting_fiscal_period::text, 2, '0')) in
+        (select max(fyp) from closed_periods) and sum(faba.gross_outlay_amount_by_award_cpe) > 0)
+    select
+        faba.disaster_emergency_fund_code,
+        coalesce(ffy.prior_fys_outlay, 0) + coalesce(cfy.current_fy_outlay, 0) as total_outlay,
+        sum(faba.transaction_obligated_amount) as obligated_amount
+    from eligible_file_c_records faba
+    left join fy_final_balances ffy on ffy.award_id = faba.award_id and ffy.disaster_emergency_fund_code = faba.disaster_emergency_fund_code
+    left join current_fy_balance cfy
+        on cfy.reporting_fiscal_period != 12 -- don't duplicate the year-end period's value if in unclosed period 01
+        and cfy.award_id = faba.award_id
+        and cfy.disaster_emergency_fund_code = faba.disaster_emergency_fund_code
+    where faba.award_id = '{award_id}'
+    group by faba.disaster_emergency_fund_code, total_outlay;
+    """
+    results = execute_sql_to_ordered_dictionary(defc_sql.format(award_id=award_id))
+    outlay_by_code = []
+    obligation_by_code = []
+    total_outlay = 0
+    total_obligations = 0
+    for row in results:
+        total_outlay += row["total_outlay"] if row["total_outlay"] is not None else 0
```

Comment:
Wouldn't the coalesce for `total_outlay` in the SQL prevent null/None?
Additionally, it would be cleaner to add coalesce into the SQL for `total_obligations`

Reply:
yeah